src/main/java/org/jfrog/hudson/pipeline/common/docker/utils/DockerUtils.java

```diff
@@ -318,18 +323,31 @@ public static int getNumberOfDependentLayers(String imageContent) throws IOExcep
         return layersNum;
     }
 
-    public static DockerClient getDockerClient(String host) {
-        NettyDockerCmdExecFactory nettyDockerCmdExecFactory;
+    public static DockerClient getDockerClient(String host, EnvVars envVars) {
 
```

Comment:
I suggest simplifying a bit the code of this method to:
```
        Builder configBuilder = DefaultDockerClientConfig.createDefaultConfigBuilder();
        if (envVars != null) {
            if (envVars.containsKey(DefaultDockerClientConfig.DOCKER_HOST)) {
                configBuilder.withDockerHost(envVars.get(DefaultDockerClientConfig.DOCKER_HOST));
            } else {
                if (!StringUtils.isEmpty(host)) {
                    configBuilder.withDockerHost(host);
                }
            }
            if (envVars.containsKey(DefaultDockerClientConfig.DOCKER_TLS_VERIFY)) {
                configBuilder.withDockerTlsVerify(envVars.get(DefaultDockerClientConfig.DOCKER_TLS_VERIFY));
            }
            if (envVars.containsKey(DefaultDockerClientConfig.DOCKER_CERT_PATH)) {
                configBuilder.withDockerCertPath(envVars.get(DefaultDockerClientConfig.DOCKER_CERT_PATH));
            }
        }

        DockerClientConfig config = configBuilder.build();
        return DockerClientBuilder.getInstance(config).withDockerCmdExecFactory(new NettyDockerCmdExecFactory()).build();
```

Reply:
That's slightly different. The host parameter would never be used when envVars is null. This would break CommonITestsPipeline.dockerPushTest at the moment.